src/checkers/inference/InferenceQualifierHierarchy.java

```diff
@@ -307,7 +273,7 @@ public AnnotationMirror getTopAnnotation(final AnnotationMirror am) {
             return varAnnot;
         } //else
 
-        return unqualified;
+        return inferenceMain.getRealTypeFactory().getQualifierHierarchy().getTopAnnotations().iterator().next();
```

Comment:
Yes, but on "new" line 58 above you have `"There should be only 1 top qualifier ( checkers.inference.qual.VarAnnot ).\n"`
This method determines the top qualifier, so shouldn't the answer also be `VarAnnot` if `am` is from the real hierarchy?